src/hotspot/cpu/aarch64/register_aarch64.cpp

```diff
@@ -49,7 +53,7 @@ const char* RegisterImpl::name() const {
 }
 
 const char* FloatRegisterImpl::name() const {
-  const char* names[number_of_registers] = {
+  static const char *const names[number_of_registers] = {
```

Comment:
While reading this code I noticed that this method is sensitive to changes to is_valid, therefore care has to be taken when changing its semantics. Or, maybe just extend the array to number_of_declared_registers and add strings for ZR and SP, just to be safe.

Reply:
Great catch! I'll have a look at how this function is used. It's a little bit awkward that there is not a simple 1-to-1 mapping between register numbers and register names.

Reply:
I understood the function as "return the functional name of the register, if there is one, otherwise return "r number".

Reply:
Yes, but the problem is that Register 31 has more than one name. I guess "r31" is the best we can do.

Reply:
Done.